src/librepgp/stream-common.cpp

```diff
@@ -400,14 +401,72 @@ file_src_close(pgp_source_t *src)
     }
 }
 
+static rnp_result_t
+init_fd_src(pgp_source_t *src, int fd, uint64_t *size)
+{
+    if (!init_src_common(src, sizeof(pgp_source_file_param_t))) {
+        return RNP_ERROR_OUT_OF_MEMORY;
+    }
+
+    pgp_source_file_param_t *param = (pgp_source_file_param_t *) src->param;
+    param->fd = fd;
+    src->read = file_src_read;
+    src->close = file_src_close;
+    src->type = PGP_STREAM_FILE;
+    src->size = size ? *size : 0;
+    src->knownsize = !!size;
+
+    return RNP_SUCCESS;
+}
+
+#ifdef _WIN32
+rnp_result_t
+init_file_src_w(pgp_source_t *src, const wchar_t *path)
```

Comment:
If we'll implent `std::string rnp_readdir_name()` as previous comment mentioned, we do not need this as we may safely call `rnp_open()` and have less system-dependent code/ifdefs.

Reply:
This is a shortcut, so we don't need to pack wide chart to utf-8 and unpack it back to potentially make the program less reliable

Reply:
@rrrooommmaaa I would still suggest to have less and more readable code but slightly more CPU cycles and allocation(s), as this is not a performance-critical code piece.

Reply:
@ni4 Done, should I delete `init_file_src_w` or let it stay?

Reply:
@rrrooommmaaa As that code is not used I think it is better to delete it. Feel free to get this PR out of the Draft so it may be approved.